infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java

```diff
@@ -474,6 +574,19 @@ private boolean isCurrentLayoutSlotUnFilled(PollReport pollReport) {
         return result;
     }
 
+    private boolean isPrimarySequencerBootstrapped(PollReport pollReport) {
+        Layout layout = serverContext.getManagementLayout();
+        String primarySequencer = layout.getSequencers().get(0);
+        if (layout.getEpoch() != pollReport.getPollEpoch()) {
+            log.warn("Received poll report for epoch {} but currently at epoch {}",
+                    pollReport.getPollEpoch(), layout.getEpoch());
+            return true;
+        }
+        NodeView nodeView = pollReport.getNodeViewMap().get(primarySequencer);
+        return nodeView.getServerMetrics().equals(ServerMetrics.getDefaultInstance())
```

Comment:
I do not think the boolean return is enough. It is leading to lot's of overloading of different things. It can lead to bugs.

Reply:
Done.